llvm/lib/SYCLLowerIR/GlobalOffset.cpp

```diff
@@ -59,11 +59,21 @@ ModulePass *llvm::createGlobalOffsetPassLegacy() {
   return new GlobalOffsetLegacy();
 }
 
+// Recursive helper function to collect Loads from GEPs in a BFS fashion.
+static void getLoads(Instruction *P, SmallVectorImpl<Instruction *> &Traversed,
```

Comment:
nit: it was a bit confusing with the name 'Traversed'. Can we not just call this also as 'PtrUses'?
Also, the function name seems a bit incomplete. May be getLoadsAndGEPs?

Reply:
I prefer the name traversed here honestly, since it is capturing the approach that this function follows a bit better.
It does a BFS for finding `Loads` and `GEP`s while keeping track of the already traversed functions.
The name focuses on the goal of this function: To collect the Loads in a small vector.
The traversal of GEPs could be more seen as a byproduct.